source/fswatch.d

```diff
@@ -296,21 +313,28 @@ struct FileWatch
 				while (true)
 				{
 					auto info = cast(inotify_event*)(eventBuffer.ptr + i);
-					assert(info.wd == wd);
 					// contains \0 at the end otherwise
 					string fileName = info.name.ptr.fromStringz().idup;
+					string absoluteFileName = directoryMap[info.wd] ~ "/" ~ fileName;
+					string relativeFilename = absoluteFileName[path.length + 1 .. $];

```

Comment:
please don't do filename manipulation like this, it is really error prone if one of the paths is not normalized. Use `std.path.relativePath` instead